buildkite/resource_pipeline.go

```diff
@@ -295,37 +417,60 @@ func DeletePipeline(ctx context.Context, d *schema.ResourceData, m interface{})
 	return nil
 }
 
-// As of August 7th 2020, GraphQL Pipeline is lacking support for updating properties:
+// As of March 9th 2021, GraphQL Pipeline is lacking support for reading properties:
 // - branch_configuration
-// - github provider configuration
+// - github provider_settings configuration
 // We fallback to REST API
-func updatePipelineWithRESTfulAPI(d *schema.ResourceData, client *Client) error {
-	slug := d.Get("slug").(string)
-	log.Printf("Updating pipeline %s ...", slug)
 
-	payload := map[string]interface{}{
-		"branch_configuration": d.Get("branch_configuration").(string),
-	}
+// PipelineExtraInfo is used to manage pipeline attributes that are not exposed via GraphQL API.
+type PipelineExtraInfo struct {
+	BranchConfiguration string `json:"branch_configuration"`
+	Provider            struct {
+		Settings struct {
+			TriggerMode                             string `json:"trigger_mode"`
+			BuildPullRequests                       bool   `json:"build_pull_requests"`
+			PullRequestBranchFilterEnabled          bool   `json:"pull_request_branch_filter_enabled"`
+			PullRequestBranchFilterConfiguration    string `json:"pull_request_branch_filter_configuration"`
+			SkipPullRequestBuildsForExistingCommits bool   `json:"skip_pull_request_builds_for_existing_commits"`
+			BuildPullRequestReadyForReview          bool   `json:"build_pull_request_ready_for_review"`
+			BuildPullRequestForks                   bool   `json:"build_pull_request_forks"`
+			PrefixPullRequestForkBranchNames        bool   `json:"prefix_pull_request_fork_branch_names"`
+			BuildBranches                           bool   `json:"build_branches"`
+			BuildTags                               bool   `json:"build_tags"`
+			CancelDeletedBranchBuilds               bool   `json:"cancel_deleted_branch_builds"`
+			PublishCommitStatus                     bool   `json:"publish_commit_status"`
+			PublishBlockedAsPending                 bool   `json:"publish_blocked_as_pending"`
+			PublishCommitStatusPerStep              bool   `json:"publish_commit_status_per_step"`
+			SeparatePullRequestStatuses             bool   `json:"separate_pull_request_statuses"`
+		} `json:"settings"`
+	} `json:"provider"`
+}
 
-	jsonStr, err := json.Marshal(payload)
+func getPipelineExtraInfo(d *schema.ResourceData, m interface{}, slug string) (*PipelineExtraInfo, error) {
+	client := m.(*Client)
+	pipelineExtraInfo := PipelineExtraInfo{}
+	err := client.makeRequest("GET", fmt.Sprintf("https://api.buildkite.com/v2/organizations/%s/pipelines/%s", client.organization, slug), nil, &pipelineExtraInfo)
```

Comment:
maybe we should move `https://api.buildkite.com/v2/organizations/%s/` part of the URL to `client.makeRequest` method, so we don't need to call `client.organization` here and duplicate the URL twice in this file.

Reply:
Or maybe just use https://github.com/buildkite/go-buildkite ? I'm wondering if there are any plans internally for adding `provider_settings` and `branch_configuration` to GraphQL API?